auth/basics/ios.md

```diff
@@ -144,110 +221,277 @@ Of course, you also would want to allow users to log out of their accounts by us
+SKYRecord *record = container.auth.currentUser;
+record[@"username"] = @"new-username";
+record[@"email"] = @"new-email@example.com";
+[container.publicCloudDatabase saveRecord:record
+                               completion:^(SKYRecord *record, NSError *error) {
+                                   if (error != nil) {
+                                       NSLog(@"Error: %@", error);
+                                       return;
+                                   }
+                                   NSLog(@"%@", record); // updated user record
+                                   console.log('Username is changed to: ', record["username"]);
+                                   [container.auth getWhoAmIWithCompletionHandler:nil];
+                               }];
+```
+
+```swift
+let container = SKYContainer.default()
+let record = container.auth.currentUser
+record?.setObject("new-username", forKey: "username" as NSCopying)
+record?.setObject("new-email@example.com", forKey: "email" as NSCopying)
+container.publicCloudDatabase.save(record!) { (record, error) in
+    if error != nil {
+        print(error!);
     }
+    print(record!);
+    print("Username is changed to: \(record?["username"] ?? "")")
+    container.auth.getWhoAmI(completionHandler: nil)
 }
-operation.container = [SKYContainer defaultContainer];
-[[SKYContainer defaultContainer] addOperation:operation];
+```
+
+## Updating a user's password
+
+The currently logged-in user can change his/her own password.
+This can be done using the [`setNewPassword:oldPassword:completionHandler:`] function.
+
+If the current password is incorrect, the SDK will return an
+`InvalidCredentials` error.
+
+```obj-c
+SKYContainer *container = [SKYContainer defaultContainer];
+[[container auth] setNewPassword:@"newPassword"
+                     oldPassword:@"oldPassword"
+               completionHandler:^(SKYRecord *user, NSError *error) {
+    if (error != nil) {
+        NSLog(@"can't change password: %@", error);
+        // Can be old password not matched?
+        return;
+    }
+
+    NSLog(@"password change successfully");
+}];
 ```
 
 ```swift
-let operation = SKYQueryOperation.queryUsersOperation(byEmails: ["john.doe@example.com", "jane.doe@example.com"])
-operation?.queryRecordsCompletionBlock = { (users, queryCursor, error) in
-    for user in users as! [SKYUser] {
-        // do something with the user
+SKYContainer.default().auth.setNewPassword("newPassword", oldPassword: "oldPassword") { (user, error) in
+    if error != nil {
+        print("can't change password: \(error)")
+        // Can be old password not matched?
+        return
     }
+
+    print("password changed successfully")
 }
-operation?.container = SKYContainer.default()
-SKYContainer.default().add(operation)
 ```
+
+Note: Changing the password of the current user will not cause
+[`SKYContainerDidChangeCurrentUserNotification`] notification to be posted.
+
+## Resetting password
```

Comment:
@tensiuyan Yes, they work together as one feature. Forgot password provides the data you can call with reset password, and after calling reset password, the user would be able to login again.